pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -779,8 +777,14 @@ func ProcessGetMetricTimeSeriesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 func buildMetricQueryFromFormulaAndQueries(formula string, queries map[string]string) (string, error) {
 
 	finalSearchText := formula
+	for key := range queries {
+		placeholder := fmt.Sprintf("${%s}", key)
+		finalSearchText = strings.ReplaceAll(finalSearchText, key, fmt.Sprintf("%v", placeholder))
+	}
+
 	for key, value := range queries {
-		finalSearchText = strings.ReplaceAll(finalSearchText, key, fmt.Sprintf("%v", value))
+		placeholder := fmt.Sprintf("${%s}", key)
+		finalSearchText = strings.ReplaceAll(finalSearchText, placeholder, fmt.Sprintf("%v", value))
 	}
```

Comment:
Yes, as you have mentioned, it still won't solve for the case `one key was a and another was aa or aba`, but those keys will not be used. I also informed Sonam regarding the same.

Sonam was using a, b, c as the individual query id's and so in the formula it will be `a + b + c`. But without placeholder if I first replace the a with the respective query and then again when I try replace b with respective query some other b characters in the replaced a query are also getting affected. 

So to fix that I created this placedholder, which is unique and usually the promql queries do not have template strings like this. So they will be unique.

Reply:
Hmm if having `${a}` inside part of a query is valid, we may have to fix this eventually anyways after someone runs a formula like that and gets an issue. What if you use the `regexp.ReplaceAllStringFunc()` like in this example: https://go.dev/play/p/8uc_wCdm-0i (fyi I just described the problem to chatgpt and it gave me this code)

I think then the replacement will always work how we want it to (at least for single-character keys)

Reply:
Yes, I think this approach is the best. I have updated the PR with this approach.